modify_model.py

```diff
@@ -0,0 +1,211 @@
+        required=True,
+        type=str,
+        help="path to trained frozen model",
+    )
+    parser.add_argument(
+        "-o",
+        "--output_path",
+        default="",
+        type=str,
+        help="path to save tf trained model",
+    )
+    parser.add_argument(
+        "-b",
+        "--batch_size",
+        type=int,
+        default=1,
+        help="required batch size for inference. Possible values = {1, 2, 3, 4}",
+    )
+    parser.add_argument(
+        "-s",
+        "--input_size",
+        type=int,
+        default=416,
+        help="input image size of trained model",
+    )
+    parser.add_argument(
+        "-i",
+        "--image_dir",
+        type=str,
+        default="",
+        help="Image directory.",
+    )
+    parser.add_argument(
+        "--category_index",
+        type=int,
+        default=0,
+        help="class index to predict",
+    )
+    parser.add_argument(
+        "--iou",
+        type=float,
+        default=0.45,
+        help="iou threshold",
+    )
+    parser.add_argument(
+        "--score",
+        type=float,
+        default=0.0,
+        help="score threshold",
+    )
+    args = parser.parse_args()
+
+    if args.output_path and not os.path.exists(args.output_path):
+        os.makedirs(args.output_path)
+
+    print(f"\n\t loading input path model... \n")
+    loaded_model = tf.keras.models.load_model(args.input_model_path)
+    print(f"\n\t loading input model successful \n")
+
+    custom_config = loaded_model.get_config()
+    custom_config["layers"][0]["config"]["batch_input_shape"] = (
+        args.batch_size,
+        args.input_size,
+        args.input_size,
+        3,
+    )
+
+    # create a custom model from loaded_model with custom batch_size
+    print(f"\n\t transfering weights to toy model from input model... \n")
+    toy_model = loaded_model.__class__.from_config(custom_config, custom_objects={})
+    weights = [layer.get_weights() for layer in loaded_model.layers[1:]]
+    for layer, w in zip(toy_model.layers[1:], weights):
+        layer.set_weights(w)
+    print(f"\n\n\n\t transfering weights to toy model successful \n\n\n")
+
+    final_model = create_yolo_model(args)
+    print(f"\n\t transfering weights to final model from toy model... \n")
+    src_weights = [layer.get_weights() for layer in toy_model.layers]
+    for layer, src_w in zip(final_model.layers, src_weights):
+        layer.set_weights(src_w)
+    print(f"\n\n\n\t transfering weights to final model successful \n\n\n")
+
+    # print(f"\n\n\t /****************final_model summary****************/ \n\n")
+    # final_model.summary()
+    print(f"\n\n\n\t saving final_model ... \n\n\n")
+    if args.output_path:
+        final_model.save(args.output_path)
+    else:
+        output_path = os.path.join(
+            os.getcwd(),
+            args.input_model_path,
+            f"yolo_batchsize_{args.batch_size}",
+        )
+        final_model.save(output_path)
```

Comment:
Overly complex, only one final_model.save needed
```suggestion
    if not args.output_path:
        output_path = os.path.join(
            os.getcwd(),
            args.input_model_path,
            f"yolo_batchsize_{args.batch_size}",
        )
    final_model.save(output_path)
```

Reply:
corrected as per suggestion.